api/collection_test.go

```diff
@@ -151,6 +203,52 @@ func TestGetCollections_collectionStoreError(t *testing.T) {
 	})
 }
 
+func TestGetCollections_invalidOrderByError(t *testing.T) {
+
+	Convey("Given an invalid orderBy value", t, func() {
+
+		paginator := &mock.PaginatorMock{
+			ReadPaginationParametersFunc: func(r *http.Request) (int, int, error) {
+				return 0, 0, nil
+			},
+		}
+
+		collectionStore := &mock.CollectionStoreMock{
+			GetCollectionsFunc: func(ctx context.Context, offset int, limit int, orderBy collections.OrderBy) ([]models.Collection, int, error) {
+				return nil, 0, errors.New("store error")
+			},
+		}
+
+		Convey("When the request is made to GET collections", func() {
+
+			r := httptest.NewRequest("GET", "http://localhost:26000/collections?order_by=fubar", nil)
+			w := httptest.NewRecorder()
+
+			api := api.Setup(context.Background(), mux.NewRouter(), paginator, collectionStore)
+			api.GetCollectionsHandler(w, r)
+
+			Convey("Then the paginator is called to extract pagination parameters", func() {
+				So(len(paginator.ReadPaginationParametersCalls()), ShouldEqual, 1)
+				So(paginator.ReadPaginationParametersCalls()[0].R, ShouldEqual, r)
+			})
+
+			Convey("Then the expected error code is returned", func() {
+				So(w.Code, ShouldEqual, http.StatusBadRequest)
+			})
+
+			Convey("Then the response body should contain the expected error response", func() {
+				body, err := ioutil.ReadAll(w.Body)
+				So(err, ShouldBeNil)
+				response := models.ErrorsResponse{}
+				err = json.Unmarshal(body, &response)
+				So(err, ShouldBeNil)
+				So(len(response.Errors), ShouldEqual, 1)
+				So(response.Errors[0].Message, ShouldEqual, collections.ErrInvalidOrderBy.Error())
+			})
+		})
+	})
+}
+
```

Comment:
Can you add a test to cover order empty/not supplied. I can see that is covered in to the code but it would be good to prove it with a test case.